docs/scenarios/jdk_8_to_17_migration.md

```diff
@@ -0,0 +1,395 @@
+  ![requesting_fix_2.png](../images/jdk_8_to_17_migration/requesting_fix_2.png)
+
+#### Reviewing Suggested Changes
+
+- The IDE highlights differences between the **current** and **suggested** versions.
+- The **right panel** displays:
+  - **MD5 is replaced with SHA-256** for secure hashing.
+  - **`BASE64Encoder` is replaced with `Base64.getEncoder()`** from `java.util.Base64`, which is the standard modern API.
+    ![file_diff_view.png](../images/jdk_8_to_17_migration/file_diff_view.png)
+
+#### Applying Fixes
+
+Click the `check icon` on the file in the left pane to apply the updates.
+![accept_solution.png](../images/jdk_8_to_17_migration/accept_solution.png)
+
+Kai correctly identified and resolved the issue:
+
+final output:
+
+```java
+import java.util.Base64; // Updated import for Base64
+MessageDigest sha256Digest = MessageDigest.getInstance("SHA-256"); // Updated to SHA-256
+return Base64.getEncoder().encodeToString(hash); // Updated to use Base64 from java.util
+```
+
+From the logs we notice,
+
+```
+ **Identify the Issue**: The code currently uses the MD5 hashing algorithm, which is considered outdated and insecure for cryptographic purposes. The recommendation is to switch to a more secure hashing algorithm, such as SHA-256.\n+\n+2. **Update the Hashing Algorithm**: We will replace the `MessageDigest.getInstance(\"MD5\")` call with `MessageDigest.getInstance(\"SHA-256\")`. This change will enhance the security of the hashing mechanism used in the application.\n+\n+3. **Remove Deprecated Classes**: The code uses `sun.misc.BASE64Encoder`, which is not part of the standard Java API and is not recommended for use. Instead, we can use `java.util.Base64` which is available in Java 8 and later. This will also ensure compatibility with OpenJDK 17.
+```
+
+The LLM was smart to identify that `sun.misc.BASE64Encoder` is deprecated and decided to update that as well.
+
+### 3.2 pom.xml
+
+#### Identifying Issues
+
+Konveyor analysis detected two dependency-related issues in `pom.xml`:
+
+- Line 23: Replace the `javax.` groupId with `jakarta.`.
+- Line 24: Replace `javax.-api` artifactId with `jakarta.-api`.
+
+#### Requesting a Fix
+
+We will solve both the issues together this time. Clicking on the **tool icon** next to the file name initiates the resolution process for both the issues.
+![pom_issues.png](../images/jdk_8_to_17_migration/pom_issues.png)
+![pom_fixes.png](../images/jdk_8_to_17_migration/pom_fixes.png)
+
+- Once complete, click on the `pom.xml` on the left pane to view the changes.
+
+#### Reviewing Suggested Changes
+
+- The **left panel** shows the current `pom.xml` file.
+- The **right panel** displays the suggested changes:
+  - **Updated `groupId` from `javax.servlet` → `jakarta.servlet`**.
+  - **Updated `artifactId` from `javax.servlet-api` → `jakarta.servlet-api`**.
+  - **Updated compiler target from `1.8` → `17`** to align with the Java 17 migration.
```

Comment:
With Llama 3 70B and it ended up suggesting 1.8 -> 11. I don't know if we have room here for improvements with prompts, or whether the model just performs less well with these tasks.

Reply:
I kinda anticipated that this might happen with different models from our experience. What I can do is mention I am using `gpt-4o-mini` and add in more details for other models and use your Llama 3 70B as example. wdyt?

if it sounds good, can you provide me the model details and the pom.xml changes? Thanks @jmontleon